pkg/services/ngalert/models/alert_rule.go

```diff
@@ -40,6 +40,7 @@ func (executionErrorState ExecutionErrorState) String() string {
 }
 
 const (
+	OkErrState       ExecutionErrorState = "OK"
```

Comment:
Do we need to introduce the "OK error state" in ngalerting when a base OK state already exists with the same string (on line 33)? Having two OK constants just muddies the model IMO, can we just map things to `ngmodels.OK` instead?

Reply:
Although the value of both fields is the same, they have different type. The one that is above has type NoDataState and belongs to another "enum". Yes, it does not guarantee compile-time safety but at least explicitly declare the API

Reply:
I'm not sure if this change would break the type checker - the only place this new value is used is in a context where we're immediately just converting it to a string?

I guess my question is more of - why add it to the model or Swagger doc at all when `ngmodels.OK` exists? `ngmodels.OK` seems to suggest a fully working, non-error state for an alert. It just seems like we already have a way of representing this state in ngalerting that we should re-use (but maybe there is some context that I am missing)

Reply:
The swagger definition is basically declaration of models on API layer. The declaration of models in `models` package is kinda domain+dto definition (I would like to have them separate too as it would help defining boundaries between data layer and domain logic). That's is the only reason why I declared those fields twice. Does that answer your question?